es6-shim.js

```diff
@@ -2009,6 +2016,9 @@
         capability.resolve = resolve;
         capability.reject = reject;
       };
+      // Initialize fields to inform optimizers about the object shape.
```

Comment:
shouldn't they be initialized to a noop function instead then, so that the type of the value doesn't change?